pkg/pipelines/scm/gitlab.go

```diff
@@ -0,0 +1,122 @@
+package scm
+
+import (
+	"net/url"
+	"strings"
+
+	"github.com/openshift/odo/pkg/pipelines/meta"
+	"github.com/tektoncd/triggers/pkg/apis/triggers/v1alpha1"
+	triggersv1 "github.com/tektoncd/triggers/pkg/apis/triggers/v1alpha1"
+)
+
+const (
+	gitlabPRBindingName   = "gitlab-pr-binding"
+	gitlabPushBindingName = "gitlab-push-binding"
+
+	gitlabCIDryRunFilters = "header.match('X-Gitlab-Event','Merge Request Hook') && body.object_attributes.state == 'opened' && body.project.path_with_namespace == '%s'  && body.project.default_branch == body.object_attributes.target_branch"
+
+	gitlabCDDeployFilters = "header.match('X-Gitlab-Event','Push Hook') && body.project.path_with_namespace == '%s' && body.ref.endsWith(body.project.default_branch)"
+)
+
+// GitLabRepository represents a service on a GitLab repo
+type GitLabRepository struct {
+	url  *url.URL
+	path string // GitLab repo path eg: (group/subgroup/../repo)
+}
+
+// NewGitLabRepository returns an instance of GitLabRepository
+func NewGitLabRepository(rawURL string) (*GitLabRepository, error) {
+	parsedURL, err := url.Parse(rawURL)
+	if err != nil {
+		return nil, err
+	}
+	var components []string
+	for _, s := range strings.Split(parsedURL.Path, "/") {
+		if s != "" {
+			components = append(components, s)
+		}
+	}
+	if len(components) < 2 {
+		return nil, invalidRepoPathError(rawURL)
+	}
+	components[len(components)-1] = strings.TrimSuffix(components[len(components)-1], ".git")
+	path := strings.Join(components, "/")
+	return &GitLabRepository{url: parsedURL, path: path}, nil
+
```

Comment:
Please consider refactoring to reduce copying codes from github.com.

Reply:
We had considered this option but as the URL for github and Gitlab are different, both functions have different return type

Reply:
Added functions to parse the URL and generate path